src/lib/settings/SettingsFolder.ts

```diff
@@ -0,0 +1,637 @@
+	 * // Remove a value from an array
+	 * await message.guild.settings.update({ disabledCommands: ['ping'] }, { arrayAction: 'remove' });
+	 *
+	 * @example
+	 * // Remove a value from an array of tuples ([[k1, v1], [k2, v2], ...])
+	 * const tags = message.guild.settings.get('tags');
+	 * const index = tags.findIndex(([tag]) => tag === 'foo');
+	 * await message.guild.settings.update({ tags: null }, { arrayIndex: index });
+	 */
+	public update(entries: ReadonlyAnyObject, options?: SettingsFolderUpdateOptions): Promise<SettingsUpdateResults>;
+	public async update(pathOrEntries: PathOrEntries, valueOrOptions?: ValueOrOptions, options?: SettingsFolderUpdateOptions): Promise<SettingsUpdateResults> {
+		if (this.base === null) {
+			throw new Error('Cannot update keys from a non-ready settings instance.');
+		}
+
+		if (this.base.existenceStatus === SettingsExistenceStatus.Unsynchronized) {
+			throw new Error('Cannot update keys from a pending to synchronize settings instance. Perhaps you want to call `sync()` first.');
+		}
+
+		let entries: [string, SerializableValue][];
+		if (typeof pathOrEntries === 'string') {
+			entries = [[pathOrEntries, valueOrOptions as SerializableValue]];
+			options = typeof options === 'undefined' ? {} : options;
+		} else if (isObject(pathOrEntries)) {
+			entries = objectToTuples(pathOrEntries as ReadonlyKeyedObject) as [string, SerializableValue][];
+			options = typeof valueOrOptions === 'undefined' ? {} : valueOrOptions as SettingsFolderUpdateOptions;
+		} else {
+			entries = pathOrEntries as [string, SerializableValue][];
+			options = typeof valueOrOptions === 'undefined' ? {} : valueOrOptions as SettingsFolderUpdateOptions;
+		}
+
+		return this._processUpdate(entries, options as InternalRawFolderUpdateOptions);
+	}
+
+	/**
+	 * Overload to serialize this entry to JSON.
+	 */
+	public toJSON(): SettingsFolderJson {
+		const json: SettingsFolderJson = {};
+		for (const [key, value] of super.entries()) {
+			json[key] = value instanceof SettingsFolder ? value.toJSON() : value;
+		}
+
+		return json;
+	}
+
+	/**
+	 * Patch an object against this instance.
+	 * @param data The data to apply to this instance
+	 */
+	protected _patch(data: ReadonlyAnyObject): void {
+		for (const [key, value] of Object.entries(data)) {
+			// Undefined values are invalid values, skip.
+			if (typeof value === 'undefined') continue;
+
+			// Retrieve the key and guard it, if it's undefined, it's not in the schema.
+			const childValue = super.get(key);
+			if (typeof childValue === 'undefined') continue;
+
+			if (childValue instanceof SettingsFolder) childValue._patch(value as ReadonlyAnyObject);
+			else super.set(key, value as SerializableValue);
+		}
+	}
+
+	/**
+	 * Initializes a SettingsFolder, preparing it for later usage.
+	 * @param folder The children folder of this instance
+	 * @param schema The schema that manages the folder
+	 */
+	protected _init(folder: SettingsFolder, schema: Schema | SchemaFolder): void {
```

Comment:
```suggestion
	protected init(folder: SettingsFolder, schema: Schema | SchemaFolder): void {
```

Reply:
I don't think these methods should be renamed. They are meant to be private and non usable outside of internals..